core/src/main/scala/kafka/server/ControllerServer.scala

```diff
@@ -200,6 +200,8 @@ class ControllerServer(
         CoreUtils.swallow(controllerApisHandlerPool.shutdown(), this)
       if (quotaManagers != null)
         CoreUtils.swallow(quotaManagers.shutdown(), this)
+      if (controllerApis != null)
+        CoreUtils.swallow(controllerApis.close(), this)
```

Comment:
Shouldn't this be closed right after the controllerApisHandlerPool?

Reply:
copy that.